http-gateway/service/getEvents.go

```diff
@@ -0,0 +1,53 @@
+package service
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/google/go-querystring/query"
+	"github.com/gorilla/mux"
+	"github.com/plgd-dev/cloud/http-gateway/uri"
+	"github.com/plgd-dev/cloud/resource-aggregate/commands"
+)
+
+func (requestHandler *RequestHandler) getEvents(w http.ResponseWriter, r *http.Request) {
+	var timestamp int64
+	var err error
+	t := r.URL.Query().Get(uri.TimestampQueryKey)
+	if t != "" {
+		timestamp, err = strconv.ParseInt(t, 10, 64)
+		if err != nil {
+			writeError(w, fmt.Errorf("failed to parse timestamp %v: %w", t, err))
+			return
+		}
+	}
+
+	vars := mux.Vars(r)
+	deviceID := vars[uri.DeviceIDKey]
+	href := vars[uri.ResourceHrefKey]
+	resourceID := commands.NewResourceID(deviceID, href).ToString()
```

Comment:
add check for empty device, href

Reply:
`ToString()` method checks both and return empty string if either is empty